people/service.go

```diff
@@ -0,0 +1,188 @@
+package people
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/Financial-Times/tme-reader/tmereader"
+	log "github.com/Sirupsen/logrus"
+	"github.com/boltdb/bolt"
+	"github.com/pborman/uuid"
+	"sync"
+	"time"
+)
+
+const (
+	cacheBucket = "person"
+	//uppAuthority = "http://api.ft.com/system/FT-UPP"
+	//tmeAuthority = "http://api.ft.com/system/FT-TME"
+)
+
+type peopleService interface {
+	getPeople() ([]personLink, bool)
+	getPersonByUUID(uuid string) (person, bool, error)
+	isInitialised() bool
+	shutdown() error
+}
+
+type peopleServiceImpl struct {
+	sync.RWMutex
+	repository    tmereader.Repository
+	baseURL       string
+	personLinks   []personLink
+	taxonomyName  string
+	maxTmeRecords int
+	initialised   bool
+	cacheFileName string
+	db            *bolt.DB
+}
+
+func newPeopleService(repo tmereader.Repository, baseURL string, taxonomyName string, maxTmeRecords int, cacheFileName string) peopleService {
+	s := &peopleServiceImpl{repository: repo, baseURL: baseURL, taxonomyName: taxonomyName, maxTmeRecords: maxTmeRecords, initialised: false, cacheFileName: cacheFileName}
+	go func(service *peopleServiceImpl) {
+		err := service.newInit()
+		if err != nil {
+			log.Errorf("Error while creating OrgService: [%v]", err.Error())
+		}
+	}(s)
+	return s
+}
+
+func (s *peopleServiceImpl) isInitialised() bool {
+	s.RLock()
+	defer s.RUnlock()
+	return s.initialised
+}
+
+func (s *peopleServiceImpl) shutdown() error {
+	log.Info("Shutingdowwn...")
+	if s.db == nil {
+		return errors.New("DB not open")
+	}
+	return s.db.Close()
+}
+
+func (s *peopleServiceImpl) getPeople() ([]personLink, bool) {
+	if len(s.personLinks) > 0 {
+		return s.personLinks, true
+	}
+	return s.personLinks, false
+}
+
+func (s *peopleServiceImpl) getPersonByUUID(uuid string) (person, bool, error) {
+	var cachedValue []byte
+	err := s.db.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket([]byte(cacheBucket))
+		if bucket == nil {
+			return fmt.Errorf("Bucket %v not found!", cacheBucket)
+		}
+		cachedValue = bucket.Get([]byte(uuid))
+		return nil
+	})
+
+	if err != nil {
+		log.Errorf("ERROR reading from cache file for [%v]: %v", uuid, err.Error())
+		return person{}, false, err
+	}
+	if len(cachedValue) == 0 {
+		log.Infof("INFO No cached value for [%v]", uuid)
+		return person{}, false, nil
+	}
+	var cachedOrg person
+	err = json.Unmarshal(cachedValue, &cachedOrg)
+	if err != nil {
+		log.Errorf("ERROR unmarshalling cached value for [%v]: %v", uuid, err.Error())
+		return person{}, true, err
+	}
+	return cachedOrg, true, nil
+
+}
+
+func (s *peopleServiceImpl) newInit() error {
```

Comment:
why is the `new` prefix added? I'm just thinking if we ever want to reuse the method e.g. for reloading endpoint....